### PR TITLE
Enable pylint checks

### PR DESCRIPTION
**Pull request by MartinBasti**
All currently failing checks were disabled until they will be fixed, see `.pylintrc` for details.

### CODE REVIEW

File: .pylintrc [hunk 1]
```diff
@@ -0,0 +1,101 @@
...
+# --enable=similarities". If you want to run only the classes checker, but have
+# no Warning level messages displayed, use"--disable=all --enable=classes
+# --disable=W"
+enable=all,
+       python3
+disable=I,
+        C,  # code style checks
+        R,  # refactoring checks
+        abstract-method,  # nice to have
+        anomalous-backslash-in-string,  # should be enabled
```
Comment (lcarva) [inline .pylintrc]: Is the goal to eventually change the code so we can enable some of these checks?

File: tests/plugins/test_pulp.py [hunk 2]
```diff
@@ -38,6 +38,11 @@
     from tests.docker_mock import mock_docker
 
 
+# py2/3 compatibility
+if sys.version_info.major > 2:
+    unicode = str
```
Comment (lcarva) [inline tests/plugins/test_pulp.py]: Can we use python six here?

File: integration-tests/test_pulp.py [hunk 3]
```diff
@@ -28,7 +28,10 @@
 
 from atomic_reactor.inner import DockerBuildWorkflow
 from atomic_reactor.util import ImageName
+# TODO: PulpUploader doesn't exist, remove or fix this test?
```
Comment (lcarva) [inline integration-tests/test_pulp.py]: Oh my... we should probably just get rid of this folder completely. @twaugh ?

File: .pylintrc (hunk 1 above)
Comment (MartinBasti) [reply .pylintrc]: It may be long term goal. Now goal is to enable pylint to not introduce new type of errors

Comment (MartinBasti) [reply .pylintrc]: At least we have to fix py2/3 compatibility errors

File: tests/plugins/test_pulp.py (hunk 2 above)
Comment (MartinBasti) [reply tests/plugins/test_pulp.py]: Done

File: integration-tests/test_pulp.py (hunk 3 above)
Comment (twaugh) [reply integration-tests/test_pulp.py]: Yes, let's remove it.

Comment (lcarva) [reply integration-tests/test_pulp.py]: Feel free to open another PR to remove it.

**Comment by lcarva** [comment]
Not updating release notes draft because this is a change solely to unit tests.

**Pull request merged**